File: point.py
class Point :
    def __init__(self, abscisse = 0, ordonné = 0)  :
        
        self.abscisse = abscisse
        self.ordonné = ordonné


    def get_abscisse(self):
        return self.abscisse
    
    def set_abscisse(self, valeur):
        self.abscisse = valeur

    def get_ordonné(self):
        return self.ordonné
    
    def set_ordonné(self, valeur):
        self.ordonné = valeur

    def distance(self,point):
        return ((self.abscisse - point.abscisse)**2 + (self.ordonné - point.ordonné)**2)**0.5

    def norm(self):
        return ((self.abscisse)**2 + (self.ordonné)**2)**0.5
    
point1 = Point(4,3 )

print("Distance:", point1.distance(Point(0, 0)))

print("Norm:", point1.norm())

File: point.java
public class Point {
    private double abscisse;
    private double ordonné;

    public Point() {
        this.abscisse = 0;
        this.ordonné = 0;
    }

    public Point(double abscisse, double ordonné) {
        this.abscisse = abscisse;
        this.ordonné = ordonné;
    }

    public double getAbscisse() {
        return abscisse;
    }

    public void setAbscisse(double valeur) {
        this.abscisse = valeur;
    }

    public double getOrdonné() {
        return ordonné;
    }

    public void setOrdonné(double valeur) {
        this.ordonné = valeur;
    }

    public double distance(Point point) {
        return Math.sqrt(Math.pow(this.abscisse - point.abscisse, 2) + Math.pow(this.ordonné - point.ordonné, 2));
    }

    public double norm() {
        return Math.sqrt(Math.pow(this.abscisse, 2) + Math.pow(this.ordonné, 2));
    }

    public static void main(String[] args) {
        Point point1 = new Point(4, 3);

        System.out.println("Distance: " + point1.distance(new Point(0, 0)));
        System.out.println("Norm: " + point1.norm());
    }
}
